### ISSUE TITLE
Vectorized DP-optimizers do not seem to learn (accuracy stays at 10%)

### ISSUE THREAD

**Issue by fraboeni** [tensorflow/privacy#162]
Hello everyone,

I am trying to run the vectorized versions of the DP optimizers (`VectorizedDPSGD` and `VectorizedDPKerasSGDOptimizer`) under the following setup:

tensorflow: 2.4.1
tensorflow-privacy: 0.5.1

Unfortunately, the learning does not seem successful.
Building the optimizer with the following code
```
#optimizer = VectorizedDPSGD(
optimizer = VectorizedDPKerasSGDOptimizer(
        l2_norm_clip=L2_NORM_CLIP,
        noise_multiplier=NOISE,
        num_microbatches=MICROBATCHES,
        learning_rate=LEARNING_RATE)
    loss = tf.keras.losses.SparseCategoricalCrossentropy(from_logits=True, reduction=tf.losses.Reduction.NONE)
```

I observe a learning performance of 10%, and the loss does not decrease over any epoch.
```
153/240 [==================>...........] - ETA: 11s - loss: 2.3164 - acc: 0.0859
```

With the non-vectorized versions, learning is successful.

**Comment by schien1729** [comment]
Hi @fraboeni ,

Sorry to hear about this. Is this with one of our tutorials?

Thanks,
Steve

**Comment by Mrxulovemingming** [comment]
I have slove this problem,Please update the tensorflow to 2.4.0.Then it solved.When I use the tensorflow 2.2.0.The accuracy stay at 10%

**Comment by fraboeni** [comment]
@Mrxulovemingming thank you for the hint. I went down from TF 2.4.1 to 2.4.0 and now it seems to work. At least accuracy goes up and the gradient clipping and noise addition are performed. Therefore, I assume the issue is solved.
